plugins/synaptics-rmi/fu-synaptics-rmi-v5-device.c

```diff
@@ -151,6 +320,22 @@ fu_synaptics_rmi_v5_device_write_firmware (FuDevice *device,
 		g_prefix_error (error, "not idle: ");
 		return FALSE;
 	}
+	if (fu_synaptics_rmi_firmware_get_sig_size (rmi_firmware) == 0 &&
+	    fu_synaptics_rmi_device_get_sig_size (self) != 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "device secure but firmware not secure");
+		return FALSE;
+	}
+	if (fu_synaptics_rmi_firmware_get_sig_size (rmi_firmware) != 0 &&
+	    fu_synaptics_rmi_device_get_sig_size (self) == 0) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "device not secure but firmware secure");
```

Comment:
How does this happen?  Pre-production?  Or should never happen?  Do we want a `--force` for this scenario maybe?

Reply:
It should never happen, but if you try to flash the wrong file (e.g. install-blob on the device) you hit this. We could add --force but I think it's probably never a good idea.

Reply:
OK agree, then no --force.